### PR TITLE
Brand of cthugha (1) (4)

### PR DESCRIPTION
**Pull request by AlvaroRF5**
Let me know what you think!

### CODE REVIEW

File: backend/arkham-core/library/Arkham/Asset/Cards/BrandOfCthugha1.hs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            [ skillTestModifier (toAbilitySource attrs 1) iid (SkillModifier sType 1)
+            , chooseFightEnemy iid (toAbilitySource attrs 1) sType
+            ]
+          | sType <- [#willpower, #combat]
+          ]
+      pure a
+    PassedThisSkillTestBy iid (isAbilitySource attrs 1 -> True) n -> do
+      player <- getPlayer iid
+      pushWhen (n == 0) $ LoseActions iid (toAbilitySource attrs 1) 1
+      push $ chooseAmounts player "Amount of Charges to Spend" (MaxAmountTarget 2) [("Charges", (1, 2))] attrs
```
Comment (halogenandtoast) [inline backend/arkham-core/library/Arkham/Asset/Cards/BrandOfCthugha1.hs:37]: We'd want to make sure you have the charges to spend, so to run this message we want to ensure there is at least one charge, and set a max equal to the minimum number between 2 and the remaining charge count.

File: backend/arkham-core/library/Arkham/Asset/Cards/BrandOfCthugha4.hs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            [ skillTestModifier (toAbilitySource attrs 1) iid (SkillModifier sType 2)
+            , chooseFightEnemy iid (toAbilitySource attrs 1) sType
+            ]
+          | sType <- [#willpower, #combat]
+          ]
+      pure a
+    PassedThisSkillTestBy iid (isAbilitySource attrs 1 -> True) n -> do
+      player <- getPlayer iid
+      pushWhen (n == 0) $ LoseActions iid (toAbilitySource attrs 1) 2
+      push $ chooseAmounts player "Amount of Charges to Spend" (MaxAmountTarget 3) [("Charges", (1, 3))] attrs
```
Comment (halogenandtoast) [inline backend/arkham-core/library/Arkham/Asset/Cards/BrandOfCthugha4.hs:37]: same thing here as the other card.

**Review by halogenandtoast — CHANGES_REQUESTED** [review]
I think as written these cards won't handle this FAQ:

> Q: [Brand of Cthugha](https://arkhamdb.com/card/08090) states " ...If you succeed, spend 1 or 2 charges. Instead of its standard damage, this attack deals 1 damage for each charge spent..." If I miss an attack and play [Oops!](https://arkhamdb.com/card/02113), how much damage does that deal? 0 because I couldn't spend a charge? What happens if I miss against an enemy engaged with another investigator, do I deal 0 or 1 damage to the engaged investigator? Other cards like the Shotgun specify the damage in case of failure, but does that apply to Oops? If I fail by 4 and the enemy is engaged with another investigator and play Oops (2), how much damage is dealt to the enemy? A: You have to be able to spend charges from Brand of Cthugha to deal damage with it. If you fail the Fight test on Brand, you cannot deal any damage—you wouldn’t deal damage to the attacked enemy, or the investigator engaged with that enemy, even if you played an Oops card.

So I'd probably want to write a test case to ensure this outcome. Maybe the easiest thing to do would be to have a modifier of `NoStandardDamage` and use `HasModifiers` to specify that. Then update the areas we check successful fights to 0 out damage when that modifier is in effect. Then you would want to change your `DamageDealt` modifiers here from `n-1` to just `n`.

If you feel up to making these changes let me know, otherwise I can take it across the finish line.

**Comment by AlvaroRF5** [comment]
I totally agree with you! I will try to implement these changes in some days since I'm currently traveling.
Thanks for the review :)

**Comment by AlvaroRF5** [comment]
I've tried to implement the changes you suggested but I feel they are a bit too complicated for me at the moment. So feel free to take this to the finish line!

**Comment by halogenandtoast** [comment]
No problem, I made the adjustments, thanks again!

**Pull request merged**